tensorflow/compiler/xla/service/gpu/amdgpu_compiler.cc

```diff
@@ -139,436 +135,101 @@ string GetROCDLDir(const HloModuleConfig& config) {
-    diag_info.print(*printer);
-  };
-  llvm_context.setDiagnosticHandlerCallBack(DiagnosticHandler, &printer);
-
-  llvm::Module llvm_module(module->name().c_str(), llvm_context);
-  // Set the target triple and the data layout.
-  llvm_module.setTargetTriple(amdgpu::kTargetTriple);
-  llvm_module.setDataLayout(amdgpu::kDataLayout);
-
-  // Determine the HLO schedule, which is an ordering of HLO instructions.  This
-  // is used by buffer assignment to enable buffer reuse, and the same ordering
-  // must also be used to determine the thunk launch schedule.
-  std::unique_ptr<StreamAssignment> stream_assignment = AssignStreams(*module);
-  TF_ASSIGN_OR_RETURN(
-      std::unique_ptr<GpuHloSchedule> hlo_schedule,
-      GpuHloSchedule::Build(*module, *stream_assignment, pointer_size_));
-
-  // Run buffer analysis on the HLO graph. This analysis figures out which
-  // temporary buffers are required to run the computation.
-  TF_ASSIGN_OR_RETURN(
-      std::unique_ptr<BufferAssignment> buffer_assignment,
-      BufferAssigner::Run(
-          module.get(), hlo_schedule->ConsumeHloOrdering(),
-          BufferSizeBytesFunction(),
-          /*color_alignment=*/
-          [](LogicalBuffer::Color) { return kXlaAllocatedBufferAlignBytes; },
-          /*allocate_buffers_for_constants=*/true));
-  DumpHloModuleIfEnabled(*module, *buffer_assignment, "after_optimizations");
-
-  IrEmitterContext ir_emitter_context(module.get(), buffer_assignment.get(),
-                                      stream_exec->platform(), &stream_exec->GetDeviceDescription(),
-                                      &llvm_module);
-
-  HloComputation* entry_computation = module->entry_computation();
-
-  IrEmitterUnnested ir_emitter(module->config(), entry_computation,
-                               &ir_emitter_context);
-
-  TF_RETURN_IF_ERROR(ir_emitter.EmitConstantGlobals());
-
-  {
-    XLA_SCOPED_LOGGING_TIMER("AMDGPUCompiler::RunBackend - IR emission");
-    TF_RETURN_IF_ERROR(entry_computation->Accept(&ir_emitter));
-  }
+namespace {
+absl::optional<bool> CanShareBufferHint(const HloInstruction* user,
+                                        const HloInstruction* operand,
+                                        const ShapeIndex& user_index) {
+  return absl::nullopt;
+}
 
-  if (user_pre_optimization_hook_) {
-    user_pre_optimization_hook_(llvm_module);
-  }
-  string ir_module_string_before_opt;
-  const bool embed_ir_in_executable =
-      module->config().debug_options().xla_embed_ir_in_executable();
-  if (VLOG_IS_ON(3) || embed_ir_in_executable) {
-    ir_module_string_before_opt = llvm_ir::DumpModuleToString(llvm_module);
-    VLOG(3) << "LLVM module before optimizations:";
-    XLA_VLOG_LINES(3, ir_module_string_before_opt);
-  }
+}  // namespace
 
-  const string& ir_dump_directory =
-      module->config().debug_options().xla_dump_to();
+HloDataflowAnalysis::CanShareBuffer AMDGPUCompiler::GetCanShareBuffer() {
```

Comment:
Looks like this can be put in base class as well, so we can get rid of both `AMDGPUCompiler` and `NVPTXCompiler` definitions

Reply:
will do.